aspnetcore/blazor/call-web-api.md

```diff
@@ -0,0 +1,216 @@
+
+In Blazor client-side apps, `HttpClient` is available as a service. The client's base address is set to the originating server's address. Inject an `HttpClient` instance using the `@inject` directive. If the component isn't routable and thus doesn't include the `@page` directive, import <xref:System.Net.Http> with an `@using` directive:
+
+```cshtml
+@using System.Net.Http
+@inject HttpClient Http
+```
+
+In the following examples, a Todo web API service processes create, read, update, and delete (CRUD) operations. The examples are based on a `TodoItem` class that stores the:
+
+* ID (`Id`, `long`) &ndash; Unique ID of the item.
+* Name (`Name`, `string`) &ndash; Name of the item.
+* Status (`IsComplete`, `bool`) &ndash; Indication if the Todo item is finished.
+
+```csharp
+private class TodoItem
+{
+    public long Id { get; set; }
+    public string Name { get; set; }
+    public bool IsComplete { get; set; }
+}
+```
+
+JSON helper methods send requests to a URI (a web API in the following examples) and process the response:
+
+* `GetJsonAsync` &ndash; Sends a GET request and parses the JSON response body to create an object.
+
+  In the following code, the `_todoItems` are displayed by the component. The `GetTodoItems` method is triggered when the component is finished rendering ([OnInitAsync](xref:blazor/components#lifecycle-methods)). See the sample app for a complete example.
+
+  ```cshtml
+  @using System.Net.Http
+  @inject HttpClient Http
+
+  @code {
+      private TodoItem[] _todoItems;
+
+      protected override async Task OnInitAsync() => 
+          _todoItems = await Http.GetJsonAsync<TodoItem[]>("api/todo");
+  }
+  ```
+
+* `PostJsonAsync` &ndash; Sends a POST request, including JSON-encoded content, and parses the JSON response body to create an object.
+
+  In the following code, `_newItemName` is provided by a bound element of the component. The `AddItem` method is triggered by selecting a `<button>` element. See the sample app for a complete example.
+
+  ```cshtml
+  @using System.Net.Http
+  @inject HttpClient Http
+
+  <input @bind="_newItemName" placeholder="New Todo Item" />
+  <button @onclick="@AddItem">Add</button>
+
+  @code {
+      private string _newItemName;
+
+      private async Task AddItem()
+      {
+          var addItem = new TodoItem { Name = _newItemName, IsComplete = false };
+          await Http.PostJsonAsync("api/todo", addItem);
+      }
+  }
+  ```
+
+* `PutJsonAsync` &ndash; Sends a PUT request, including JSON-encoded content.
+
+  In the following code, `_editItem` values (`Id`, `Name`, `IsCompleted`) are provided by bound elements of the component. The `SaveItem` method is triggered by selecting the Save `<button>` element. See the sample app for a complete example.
+
+  ```cshtml
+  @using System.Net.Http
+  @inject HttpClient Http
+
+  <input @bind="_editItem.Id" />
```

Comment:
Remove? It's a bit unusually I think for the identity of an entity to be user editable.

Reply:
It was just a simplification over what the sample app does. I'll make this more like the sample app on the next commit.